src/wasm/wasm-binary.cpp

```diff
@@ -2963,19 +2961,18 @@ void WasmBinaryReader::pushExpression(Expression* curr) {
   if (type.isTuple()) {
     // Store tuple to local and push individual extracted values
     Builder builder(wasm);
-    // Non-nullable types require special handling as they cannot be stored to
-    // a local, so we may need to use a different local type than the original.
-    auto localType = type;
-    if (!wasm.features.hasGCNNLocals()) {
-      std::vector<Type> finalTypes;
-      for (auto t : type) {
-        if (t.isNonNullable()) {
-          t = Type(t.getHeapType(), Nullable);
-        }
-        finalTypes.push_back(t);
-      }
-      localType = Type(Tuple(finalTypes));
-    }
+    // Non-nullable types require special handling as they cannot be stored to a
+    // local, so we may need to use a different local type than the original.
+    // TODO: Now that the spec supports limited non-nullable locals, is this
+    // still true?
```

Comment:
I think if it is then we have a bug with tuple handling...

Reply:
I tried deleting this special handling and everything seemed to work fine.